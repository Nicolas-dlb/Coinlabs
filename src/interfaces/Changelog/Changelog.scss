.changelog {
  padding-left: 19px;
  padding-bottom: 3vh;
  border-radius: 30px;
  z-index: 50;
  background: var(--secondary-bg-color);
  padding-top: 10vh;
  opacity: 0;
  pointer-events: none;
  margin-top: 1vh;
  overflow: hidden;
  position: relative;
  bottom: 357vh;
  height: 100%;
  width: 100%;
  .change_container {
    display: flex;
    align-items: center;
    justify-content: flex-start;
    margin: 30px 0;
    .added,
    .updated,
    .fixed {
      width: 70px;
      height: 22px;

      display: flex;
      align-items: center;
      justify-content: center;
      border-radius: 30px;
      margin-right: 10px;
      font-size: 13px;
    }
    .added {
      background: #1baa63;
    }
    .updated {
      background: #4689b6;
    }
    .fixed {
      background: #db9a67;
    }
    .added::after {
      content: "Added";
    }
    .updated::after {
      content: "Updated";
    }
    .fixed::after {
      content: "Fixed";
    }
  }
  .note_container {
    display: flex;
    align-items: center;
    justify-content: flex-start;
    margin: 30px 0;
    .point {
      width: 7px;
      height: 7px;
      border-radius: 50%;
      background: grey;
    }
    .added,
    .updated,
    .fixed {
      font-weight: bold;
      font-size: 16px;
      color: grey;
      margin: 0 10px;
    }
    .added::after {
      content: "Added";
    }
    .updated::after {
      content: "Updated";
    }
    .fixed::after {
      content: "Fixed";
    }
  }
}

@media only screen and (min-width: 992px) {
  .changelog {
    padding-left: 208px;
    bottom: 303vh;
    .title,
    .page-title {
      margin-left: 1%;
    }
  }
}
