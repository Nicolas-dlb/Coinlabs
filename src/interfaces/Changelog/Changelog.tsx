import "./Changelog.scss";

function Changelog() {
  return (
    <div id="changelog" className="changelog">
      <div className="upcoming_changes">
        <p className="page-title">v1.1 Upcoming change</p>
        <div className="change_container">
          <p className="added" />
          <p>More cryptocurrencies</p>
        </div>
        <div className="change_container">
          <p className="added" />
          <p>Day theme</p>
        </div>

        <div className="change_container">
          <p className="updated" />
          <p>Show password on login page</p>
        </div>
        <div className="change_container">
          <p className="fixed" />
          <p>Improve app load time</p>
        </div>
        <div className="change_container">
          <p className="fixed" />
          <p>Prevent app to keep log when exit</p>
        </div>
        <div className="title">Dev Notes</div>
        <div className="note_container">
          <div className="point" />
          <p className="added" />
          <p>CI/CD integration</p>
        </div>
        <div className="note_container">
          <div className="point" />
          <p className="added" />
          <p>Update changelog automaticaly</p>
        </div>
        <div className="note_container">
          <div className="point" />
          <p className="added" />
          <p>Incrase Tests number</p>
        </div>
        <div className="note_container">
          <div className="point" />
          <p className="updated" />
          <p>New Eslint configuration file</p>
        </div>
      </div>
    </div>
  );
}

export default Changelog;
