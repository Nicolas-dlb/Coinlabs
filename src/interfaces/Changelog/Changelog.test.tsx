/* eslint-disable no-unused-vars */
import { render } from "@testing-library/react";
import { mount, shallow } from "enzyme";
import React from "react";
import Changelog from "./Changelog";

describe("Changelog", () => {
  let ChangelogComponent: any;

  beforeEach(() => {
    ChangelogComponent = shallow(<Changelog />);
  });

  it("should render correctly", () => {
    expect(ChangelogComponent).toMatchSnapshot();
  });
  it("should render upcoming change", () => {
    expect(ChangelogComponent.find(".added").first().exists()).toBe(true);
  });
});
