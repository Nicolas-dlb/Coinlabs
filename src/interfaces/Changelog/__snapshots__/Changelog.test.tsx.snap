// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Changelog should render correctly 1`] = `
<div
  className="changelog"
  id="changelog"
>
  <div
    className="upcoming_changes"
  >
    <p
      className="page-title"
    >
      v1.1 Upcoming change
    </p>
    <div
      className="change_container"
    >
      <p
        className="added"
      />
      <p>
        More cryptocurrencies
      </p>
    </div>
    <div
      className="change_container"
    >
      <p
        className="added"
      />
      <p>
        Day theme
      </p>
    </div>
    <div
      className="change_container"
    >
      <p
        className="updated"
      />
      <p>
        Show password on login page
      </p>
    </div>
    <div
      className="change_container"
    >
      <p
        className="fixed"
      />
      <p>
        Improve app load time
      </p>
    </div>
    <div
      className="change_container"
    >
      <p
        className="fixed"
      />
      <p>
        Prevent app to keep log when exit
      </p>
    </div>
    <div
      className="title"
    >
      Dev Notes
    </div>
    <div
      className="note_container"
    >
      <div
        className="point"
      />
      <p
        className="added"
      />
      <p>
        CI/CD integration
      </p>
    </div>
    <div
      className="note_container"
    >
      <div
        className="point"
      />
      <p
        className="added"
      />
      <p>
        Update changelog automaticaly
      </p>
    </div>
    <div
      className="note_container"
    >
      <div
        className="point"
      />
      <p
        className="added"
      />
      <p>
        Improve tests coverage
      </p>
    </div>
    <div
      className="note_container"
    >
      <div
        className="point"
      />
      <p
        className="updated"
      />
      <p>
        New Eslint configuration file
      </p>
    </div>
  </div>
</div>
`;
